JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -211,41 +204,57 @@ private static void GetAndSetTokens(Map<String, String> formData) throws Excepti
 	    	    String res = buf.toString("UTF-8");
 	        	JsonElement jElement = new JsonParser().parse(res);
 	            JsonObject jObject = jElement.getAsJsonObject();
-	            
 				// set Tokens as System Properties - new values to access_token and refresh_token
-				System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, jObject.get("access_token").getAsString());
-				System.setProperty(OAUTH_REFRESH_TOKEN_VAR_STRING, jObject.get("refresh_token").getAsString());
-
+	            if(null == jObject.get("access_token").getAsString().isEmpty() || jObject.get("access_token").getAsString().isEmpty())
```

Comment:
In this block, instead of doing jObject.get a bunch of times, instantiate the two variables first with
String accessToken = jObject.has("access_token") ? jObject.get("access_token").getAsString() : null;
String refreshToken = jObject.has("refresh_token") ? jObject.get("refresh_token").getAsString() : null;

Then throw an exception if they are null or empty.

Reply:
Added the check as per suggestion. It will be available in next commit.